JCLOUDS-1153 Fix empty Docker repoTags field in ImageToImage function

kwart opened the pull request:
JIRA: https://issues.apache.org/jira/browse/JCLOUDS-1153

This commit allows to work with empty `repoTags` field in Docker images.

nacx commented on docker/src/main/java/org/jclouds/docker/compute/functions/ImageToImage.java:
```diff
@@ -91,9 +91,15 @@ public OsFamily apply(final String description) {
+      if (repoTag.contains(":")) {
+         version = get(Splitter.on(":").split(repoTag), 1);
+      } else {
+         version = "<none>";
```
Would it be better to use `version = repoTag`? This way we don't couple this method to the placeholder value and make the method itself more flexible.

kwart replied:
I will get rid of the whole `parseVersion` method. It's still not correct. There can be `repoTag` with port provided for repo-host (e.g. `internal-repository.company.example:8888/a/b/c/d:test`)

kwart commented:
New version was pushed. I've removed the `parseVersion` method and made the relation between the `name` and `osVersion` more clear. A new test was also added for the repoTag which contains host and port.

nacx commented:
Thanks @kwart! Pushed to master as [05a5ae2e](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/05a5ae2e).
